sympy/discrete/transforms.py

```diff
@@ -3,22 +3,33 @@
 Walsh Hadamard Transform, Mobius Transform
 """
 
-from sympy.core import S, Symbol, sympify
+from __future__ import annotations
+
+from typing import Sequence, SupportsIndex, Union
+
+from sympy.core import S, Symbol, Expr, sympify
 from sympy.core.function import expand_mul
 from sympy.core.numbers import pi, I
 from sympy.functions.elementary.trigonometric import sin, cos
 from sympy.ntheory import isprime, primitive_root
 from sympy.utilities.iterables import ibin, iterable
 from sympy.utilities.misc import as_int
 
+_NumericExpr = Union[Expr, float, complex, int]
+_NumericSequence = Sequence[_NumericExpr]
+_IntLike = SupportsIndex
+_IntSequence = Sequence[_IntLike]
```

Comment:
I don't think these aliases really help. It would be better just to use e.g. `Expr | complex` or `Sequence[SupportsIndex]` directly in the function signatures.